templates/todo/projects/nodejs-mongo-aks/.repo/bicep/infra/main.bicep

```diff
@@ -0,0 +1,106 @@
+targetScope = 'subscription'
+
+@minLength(1)
+@maxLength(64)
+@description('Name of the the environment which is used to generate a short unique hash used in all resources.')
+param environmentName string
+
+@minLength(1)
+@description('Primary location for all resources')
+param location string
+
+// Optional parameters to override the default azd resource naming conventions. Update the main.parameters.json file to provide values. e.g.,:
+// "resourceGroupName": {
+//      "value": "myGroupName"
+// }
+param applicationInsightsDashboardName string = ''
+param applicationInsightsName string = ''
+param cosmosAccountName string = ''
+param cosmosDatabaseName string = ''
+param keyVaultName string = ''
+param logAnalyticsName string = ''
+param resourceGroupName string = ''
+
+@description('Id of the user or app to assign application roles')
+param principalId string = ''
```

Comment:
Something we'll need to consider in the CI/CD case - the `principalId` will be different from what it was when one did a `provision` locally.  Is that going to cause the cluster to be replaced in a dangerous way?

Reply:
This `principalId` is no longer passed into the AKS module after the most recent refactor.